controllers/sessionController.js

```diff
@@ -0,0 +1,50 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+// const ValidationError = require("../utils/validationError");
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res) => {
+  try {
+    if (req.body.password !== req.body.password1) {
+      // throw new ValidationError("password", "The passwords do not match");
+      throw new Error("The passwords do not match.");
+    }
+    await User.create(req.body);
+  } catch (e) {
+    if (e.constructor.name === "ValidationError") {
+      parseVErr(e, req);
+    } else if (e.name === "MongoServerError" && e.code === 11000) {
+      req.flash("error", "That email address is already registered.");
+    } else {
+      req.flash("error", e.message);
+      // req.flash("error", "Unable to register that account.");
+    }
+    return res.render("register", { errors: req.flash("error") });
+  }
+  res.redirect("/");
+};
+
+const logoff = (req, res) => {
+  req.session.destroy(function (err) {
+    if (err) {
+      console.log(err);
+    }
+    res.redirect("/");
+  });
+};
+
+const logonShow = (req, res) => {
+  if (req.user) {
```

Comment:
Nice usage of an early return here! Alternatively, since it’s a single instruction for the if/else condition, you could also use the ternary operator:
```js
return req.user ? res.redirect("/") : res.render("logon");
```